src/anet.c

```diff
@@ -484,7 +484,7 @@ static int _anetTcpServer(char *err, int port, char *bindaddr, int af, int backl
 
         if (af == AF_INET6 && anetV6Only(err,s) == ANET_ERR) goto error;
         if (anetSetReuseAddr(err,s) == ANET_ERR) goto error;
-        if (anetListen(err,s,p->ai_addr,p->ai_addrlen,backlog) == ANET_ERR) goto error;
+        if (anetListen(err,s,p->ai_addr,p->ai_addrlen,backlog) == ANET_ERR) return ANET_ERR;
```

Comment:
@soloestoy You missed to call freeaddrinfo(servinfo) 
```
freeaddrinfo(servinfo);
```

Reply:
Oops... my bad, PR updated, thx~